app/scripts/controllers/transactions/index.js

```diff
@@ -331,6 +331,137 @@ export default class TransactionController extends EventEmitter {
     });
   }
 
+  // ====================================================================================================================================================
+
+  /**
+   * @param {number} txId
+   * @returns {TransactionMeta} the txMeta who matches the given id if none found
+   * for the network returns undefined
+   */
+  _getTransaction(txId) {
+    const { transactions } = this.store.getState();
+    return transactions[txId];
+  }
+
+  _validateTransaction(txId) {
+    return (
+      this._getTransaction(txId).status === TRANSACTION_STATUSES.UNAPPROVED
+    );
+  }
+
+  _updateTransaction(txParams, note) {
+    const normalizedTxParams = txUtils.normalizeTxParams(txParams);
+    const withAdditionalMeta = this.txStateManager.generateTxMeta({
```

Comment:
`generateTxMeta` is used when we are creating a new transaction object

Also, `txParams` is just one property of `txMeta` that we might need to update. And we might not need to normalize the txParams for every call to _updateTransaction

It is possible that we don't need this `_updateTransaction` method, and can instead just call `this.txStateManager.updateTransaction` in each of the other methods added here